azure_functions_worker/bindings/datumdef.py

```diff
@@ -92,6 +94,66 @@ def from_typed_data(cls, td: protos.TypedData):
 
         return cls(val, tt)
 
+    @classmethod
+    def from_rpc_shared_memory(cls, shmem: protos.RpcSharedMemory, shmem_mgr: SharedMemoryManager) -> Optional[Datum]:
+        """
+        Reads the specified shared memory region and converts the read data into a datum object of
+        the corresponding type.
+        """
+        mmap_name = shmem.name
+        offset = shmem.offset
+        count = shmem.count
+        data_type = shmem.type
+        if data_type == protos.RpcDataType.bytes:
+            val = shmem_mgr.get_bytes(mmap_name, offset, count)
+            if val is not None:
+                return cls(val, 'bytes')
+        elif data_type == protos.RpcDataType.string:
+            val = shmem_mgr.get_string(mmap_name, offset, count)
+            if val is not None:
+                return cls(val, 'string')
+        return None
+
+    @classmethod
+    def to_rpc_shared_memory(cls, datum: Datum, shmem_mgr: SharedMemoryManager) -> protos.RpcSharedMemory:
+        """
+        Writes the given value to shared memory and returns the corresponding RpcSharedMemory
+        object which can be sent back to the functions host over RPC.
+        """
+        if datum.type == 'bytes':
+            value = datum.value
+            map_name = shmem_mgr.put_bytes(value)
+            if map_name is not None:
+                shmem = protos.RpcSharedMemory(
+                    name=map_name,
+                    offset=0,
+                    count=len(value),
+                    type=protos.RpcDataType.bytes)
+                return shmem
+            else:
+                raise Exception(
+                    f'cannot write datum value (type: {datum.type}) into '
+                    f'shared memory (name: {map_name})'
+                )
+        elif datum.type == 'string':
+            value = datum.value
+            map_name = shmem_mgr.put_string(value)
+            if map_name is not None:
+                shmem = protos.RpcSharedMemory(
+                    name=map_name,
+                    offset=0,
+                    count=len(value),
+                    type=protos.RpcDataType.string)
+                return shmem
+            else:
```

Comment:
Avoid the else, as the if already returns.
(This is a typical pylint comment, you may want to run it, it will give a lot of non-sense suggestions but good to check.)